token/program-2022/src/instruction.rs

```diff
@@ -251,6 +256,10 @@ pub enum TokenInstruction {
     /// decimals value is checked by the caller.  This may be useful when
     /// creating transactions offline or within a hardware wallet.
     ///
+    /// If the accounts are `AccountWithTransferFee`s, the fee is withheld in the
+    /// source account. If either account is owned by the `fee_account_owner`,
```

Comment:
I went both ways on this one but don't have a strong opinion.  I thought it would be clearer to keep it in the source account, so that the recipient never needs to know about fees withheld.

If I see that I just received $10 in my balance, and also see $1 withheld in my account, I'm going to wonder how to access that, and then get angry when I find out about transfer fees.  But maybe that part is just never shown, so people never ask questions.

As for the additional transfer of all current withheld amount, it certainly does the sweeping in fewer transactions.  On the other hand, I'm writing a script to sweep fees from accounts, there's a chance that those fees will be gone between the time I see the withheld amount and the time I execute the transaction, and then I need to search again to see where those fees went.  That seems unintuitive to me.

Reply:
Yes good point about the sweep script. If the fee collected at the destination, one could grief the sweep script by moving the withheld amount around continually 😆

Reply:
hehe yeah I was thinking that too.  At least it would have to stop eventually since the fees will constantly be left as the amount is transferred around.

I'll change the comment to mention that the transfer fee goes to the destination, that makes more sense.